### DISCUSSION TITLE
How to download calendar file (.ics) to mobile phone (Android)?

### DISCUSSION THREAD

**Discussion by tesicg** [quasarframework/quasar#10823] [General - Components / Directives / etc]
We have Vue.js mobile app that uses Quasar framework and Cordova. The code we use looks as following:

    <template>
    ...
      <div v-cloak class="text-center q-my-sm">
        <q-btn
          label="Add to calendar"
          @click="addToCalendar"
        />
      </div>
    ...
    </template>

    <script>
    export default {
      props: {
        order: {
          type: Object,
          required: true
        },
        ...
      },
      data: () => ({
        androidPermissions: {},
        ...
      }),
      methods: {
        checkPermissionCallback(status) {
          if (!status.hasPermission) {
            const errorCallback = function() {
              console.warn('Storage permission is not turned on')
            }
            // Asking permission to the user
            this.androidPermissions.requestPermission(
              this.androidPermissions.READ_EXTERNAL_STORAGE,
              function(status) {
                if (!status.hasPermission) {
                  errorCallback()
                } else {
                  // proceed with downloading
                  this.downloadFile()
                }
              }, errorCallback)
          } else {
            this.downloadFile()
          }
        },
        downloadFile() {
          const cordovaFile = window?.cordova?.file
          let filePath = cordovaFile?.externalRootDirectory + 'download/calendar.ics'
          let fileTransfer = new window.FileTransfer()
          let uri = encodeURI(decodeURIComponent(this.getCalendarFilePath()))

          // Downloading the file
          fileTransfer.download(uri, filePath,
            function(entry) {
              console.info('Successfully downloaded file, full path is ' + entry.fullPath)
              console.info(entry)
            },
            function(error) {
              console.info('error: ', error)
            },
            false
          )
        },
        addToCalendar() {
          this.androidPermissions = window?.cordova?.plugins?.permissions
          this.androidPermissions?.checkPermission(this.permissions.READ_EXTERNAL_STORAGE, this.checkPermissionCallback, null)
        },
        baseUrl() {
          return axiosInstance.defaults.baseURL
        },
        getCalendarFilePath() {
          return `${this.baseUrl()}/api/aggregator/bookings/${this.order.id}/calendar.ics`
        },
      },
    }
    </script>

We use Cordova plugin for user permission on the following address:

https://github.com/NeoLSN/cordova-plugin-android-permissions

After uploading the app to real mobile phone, we get the following error:

[![enter image description here][1]][1]

How can we download calendar file to mobile phone with Android?

  [1]: https://i.stack.imgur.com/EKIOF.jpg

**Comment by ontwikkelfabriek** [comment]
@tesicg 

> We use Cordova plugin for user permission on the following address:

Settings permissions in some plugin? Don't know what plugin you're talking about because you don't need a plugin to set permissions.

To gain access to the Android file system you need more than just setting permissions , you need the `file plugin`:
https://cordova.apache.org/docs/en/10.x/reference/cordova-plugin-file/

Besides that , from reading some of your code, I guess what you actually want is to gain access to the Android calendar. Why not use this calendar plugin:
https://www.npmjs.com/package/cordova-plugin-calendar
https://ionicframework.com/docs/native/calendar

**Reply by tesicg** [reply]
The `file plugin` is already installed.

We used the following article as starting point:

https://stackoverflow.com/questions/49689385/cordova-how-to-move-file-to-the-download-folder

**Comment by ontwikkelfabriek** [comment]
@tesicg 

> Besides that , from reading some of your code, I guess what you actually want is to gain access to the Android calendar. Why not use this calendar plugin:

Please respond to all parts of my post. Again, why not use the android calendar API( with a Cordova plugin) instead of messing around with files?

> We used the following article as starting point:

That article is 3 years old, that's a long time in IT.  I think to know that the permission system in Android has seen changes since then.

To have any chance to solve your issue , I would suggest you create a working repo that will reproduce your problem and errors, so I can run it locally.
